datahub/event/test/test_views_v4.py

```
@@ -0,0 +1,787 @@
+from operator import itemgetter
+
```

You should add a test case for when `related_trade_agreements_exist` is True, but there are no trade agreements specified - presumably it should give a 400 response to say that `related_trade_agreements_exist` but no `related_trade_agreements` were specified.

---

If the user doesn't select anything on the frontend despite clicking 'yes' a `null` is sent back, the only time no trade agreements are sent are if 'no' is selected

---

It sounds like that is dependent on the frontend implementation stopping you do bad things, but the backend should be agnostic of that. At the moment you could end up with bad data being stored in the database, the API could be called with this input...
```
{
    has_related_trade_agreements: True,
    related_trade_agreements: [],
    ...
}
```
Which would result in non-sensical / inconsistent data being stored in the database - it should raise a 400 error to specify that this is bad input. 
Ideally you would also have test cases for the `PATCH` method to make sure that the input does not conflict with what is already stored in the database - ie when an existing event has `has_related_trade_agreements` set to True and then you PATCH with `related_trade_agreements = [] ` (or vice-versa), the data will end up being inconsistent, so this should also raise a 400 error.

---

I think Chris mentioned just putting `has_related_trade_agreements` on the serializer but not the database model - that would get over the problem of inconsistent data so it would be less important to raise 400 errors when inconsistent data is received.